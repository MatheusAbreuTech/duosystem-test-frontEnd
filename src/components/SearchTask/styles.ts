import theme from '@/styles/theme'
import styled from 'styled-components'

export const Wrapper = styled.form`
  width: 100%;
  height: 44px;
  padding-right: 1rem;
  border-radius: 5px;
  background-color: #fff;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid ${theme.colors.black};
`

export const ButtonSearch = styled.button`
  border: none;
  outline: none;
  background-color: transparent;
  cursor: pointer;
`
