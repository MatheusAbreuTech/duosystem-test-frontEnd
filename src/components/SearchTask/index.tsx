'use client'

import { useState } from 'react'
import * as S from './styles'
import Image from 'next/image'

const SearchTask = () => {
  const [inputValue, setInputValue] = useState('')
  return (
    <S.Wrapper>
      <S.Input
        type="text"
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
        placeholder="Pesquise uma tarefa..."
      />

      <Image src="/img/search.svg" alt="" width="20" height="20" />
    </S.Wrapper>
  )
}

export default SearchTask
