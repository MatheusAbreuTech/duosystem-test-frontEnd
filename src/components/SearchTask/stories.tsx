import { Meta, StoryObj } from '@storybook/react'
import SearchTask from '.'

export default {
  title: 'SearchTask',
  component: SearchTask
} as Meta

export const Default: StoryObj = {}
