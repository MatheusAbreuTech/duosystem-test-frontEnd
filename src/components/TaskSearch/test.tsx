import { render } from '@testing-library/react'

import SearchTask from '.'

describe('<SearchTask />', () => {
  it('should render the SearchTask component in screen', () => {
    // renderiza o component
    const { container } = render(<SearchTask />)

    expect(container).toBeInTheDocument()
  })
})
