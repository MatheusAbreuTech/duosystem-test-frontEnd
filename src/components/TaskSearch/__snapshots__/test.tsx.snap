// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchTask /> should render the SearchTask component in screen 1`] = `
.c0 {
  width: 100%;
  height: 44px;
  padding-right: 1rem;
  border-radius: 5px;
  background-color: #fff;
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid #383838;
}

.c2 {
  border: none;
  outline: none;
  background-color: transparent;
  cursor: pointer;
}

.c1 {
  width: 100%;
  height: 42px;
  padding: 0 1rem;
  background-color: #fff;
  outline: none;
  border: none;
  font-size: 1.6rem;
  border-radius: 5px;
}

<form
  action=""
  class="c0"
>
  <input
    class="c1"
    data-testid="searchValue"
    name="searchValue"
  />
  <button
    class="c2"
    type="submit"
  >
    <img
      alt=""
      height="20"
      src="/img/search.svg"
      width="20"
    />
  </button>
</form>
`;
